### PR TITLE
add insert function on List Fixes #402

### LINKED ISSUE #402: Feature request: insert()
For convenience: `insert()` would be nice. Basically just a convenience wrapped around `slice()`, similar to `remove()`.

### PR DESCRIPTION
**Pull request by rawls238**
@leebyron

### CODE REVIEW

File: dist/immutable.d.ts [hunk 1]
```diff
@@ -180,6 +180,7 @@ declare module Immutable {
      */
     delete(index: number): List<T>;
     remove(index: number): List<T>;
+    insert(index: number, value: T): List<T>;
```
Comment (leebyron) [inline dist/immutable.d.ts]: Can you add a docblock above this similar to the one above `delete` which describes the method in similar language to other documentation in this file?

Comment (rawls238) [inline dist/immutable.d.ts]: yup, will do

Comment (rawls238) [inline dist/immutable.d.ts]: added

**Comment by leebyron** [comment]
Nice work, thanks

**Pull request merged**